web/config.py

```diff
@@ -860,10 +877,30 @@
 if os.path.exists(system_config_dir + '/config_system.py'):
     try:
         sys.path.insert(0, system_config_dir)
-        from config_system import *
+        import config_system
+        config_system_settings = get_variables_from_module('config_system')
+        user_config_settings.update(config_system_settings)
     except ImportError:
         pass
 
+# Update settings for 'LOG_FILE', 'SQLITE_PATH', 'SESSION_DB_PATH',
+# 'AZURE_CREDENTIAL_CACHE_DIR', 'KERBEROS_CCACHE_DIR', 'STORAGE_DIR'
+# of DATA_DIR is user defined
+data_dir_dependent_settings = ['LOG_FILE', 'SQLITE_PATH', 'SESSION_DB_PATH',
+                               'AZURE_CREDENTIAL_CACHE_DIR',
+                               'KERBEROS_CCACHE_DIR', 'STORAGE_DIR']
+
+if 'DATA_DIR' in user_config_settings:
+    for setting in data_dir_dependent_settings:
+        if setting not in user_config_settings:
+            data_dir = user_config_settings['DATA_DIR']
+            file_dir_name = \
+                locals().get(setting).os.path.basename(locals().get(setting))
```

Comment:
Just one minor thing, simply `os.path.basename(locals().get(setting))` instead of `locals().get(setting).os.path.basename(locals().get(setting))` won't work?

Reply:
Accommodated review comments